src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -176,23 +176,15 @@ private void RequestRefresh(AuthenticationTicket ticket)
 
         private CookieOptions BuildCookieOptions()
         {
-            var cookieOptions = new CookieOptions
-            {
-                Domain = Options.CookieDomain,
-                SameSite = Options.CookieSameSite,
-                HttpOnly = Options.CookieHttpOnly,
-                Path = Options.CookiePath ?? (OriginalPathBase.HasValue ? OriginalPathBase.ToString() : "/"),
-            };
+            var cookieOptions = Options.Cookie.Build(Context);
+            // ignore the 'Expires' value as this will be computed elsewhere
+            cookieOptions.Expires = null;
 
-            if (Options.CookieSecure == CookieSecurePolicy.SameAsRequest)
+            if (Options.Cookie.Path == null && OriginalPathBase.HasValue)
```

Comment:
Is this doing the same thing? I'd still expect, if a cookie does have a path but I say SameAsRequest it'd match the Secure flag to the request protocol.